### PR TITLE
update pb def files for pilot/fg/drone

### PR DESCRIPTION
**Pull request by calvinyv**

### CODE REVIEW

File: api/drone/drone.proto [hunk 1]
```diff
@@ -7,24 +7,56 @@ syntax = "proto3";
 package openpitrix.drone;
 
 service DroneService {
-	rpc StartConfd (Task) returns (Empty);
+	rpc GetInfo(Empty) returns (Info);
+	rpc GetConfdConfig(Empty) returns (ConfdConfig);
+	rpc GetBackendConfig(Empty) returns (BackendConfig);
+	rpc StartConfd (Empty) returns (Empty);
```
[the remark this answers is not in the document]
Comment (calvinyv) [reply api/drone/drone.proto]: why it needs that? I thought StartConfd was called by fg by grpc on drone, drone should know itself's config.

[the remark this answers is not in the document]
Comment (chai2010) [reply api/drone/drone.proto]: drone should know the etcd nodes ip. the ConfdConfig is optional

File: api/frontgate/frontgate.proto [hunk 2]
```diff
@@ -6,39 +6,87 @@ syntax = "proto3";
...
+	string drone_ip = 1;
+	repeated string frontgate_ips = 2;
+	int32 timeout_seccond = 3;
 }
 
-message TaskDirective {
+message StopConfdRequest {
 	string drone_ip = 1;
-	string frontgate_id = 2;
+	int32 timeout_seccond = 3;
```
Comment (zheng1) [inline api/frontgate/frontgate.proto]: `timeout_seccond` typo?

File: api/frontgate/frontgate.proto [hunk 3]
```diff
@@ -6,39 +6,87 @@ syntax = "proto3";
...
 
-	rpc RegisterCmd (Task) returns (Empty);
-	rpc DeregisterCmd (Task) returns (Empty);
+	rpc RegisterCmd (RegisterCmdRequest) returns (Empty);
+	rpc DeregisterCmd (DeRegisterCmdRequest) returns (Empty);
+}
+
+service FrontgateServiceForDrone {
+	rpc GetInfo (Empty) returns (Info);
+	rpc PublicCmdResult(CommandResult) returns (Empty);
```
Comment (zheng1) [inline api/frontgate/frontgate.proto]: `PublicCmdResult` => `ReportCmdResult`

File: api/frontgate/frontgate.proto [hunk 4]
```diff
@@ -6,39 +6,87 @@ syntax = "proto3";
...
+}
+
+message CommandResult {
+	string drone_ip = 1;
+	string cmd = 2;
```
Comment (zheng1) [inline api/frontgate/frontgate.proto]: Should add the status of cmd to result

File: api/drone/drone.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (calvinyv) [reply api/drone/drone.proto]: comment addressed

File: api/frontgate/frontgate.proto (hunk 2 above)
Comment (calvinyv) [reply api/frontgate/frontgate.proto]: comment addressed

File: api/frontgate/frontgate.proto (hunk 3 above)
Comment (calvinyv) [reply api/frontgate/frontgate.proto]: comment addressed

File: api/frontgate/frontgate.proto (hunk 4 above)
Comment (calvinyv) [reply api/frontgate/frontgate.proto]: comment addressed

File: api/drone/drone.proto [hunk 5]
```diff
@@ -7,24 +7,66 @@ syntax = "proto3";
...
+	rpc GetConfdStatus(Empty) returns (ConfdStatus)
 }
 
-message Task {
-	string id = 1;
-	string action = 2;
-	string target = 3;
-
-	TaskDirective directive = 4;
+service DroneServiceForFrontGate {
```
Comment (chilianyi) [inline api/drone/drone.proto]: We can treat `Frontgate` as one word.

File: api/drone/drone.proto [hunk 6]
```diff
@@ -7,24 +7,66 @@ syntax = "proto3";
 package openpitrix.drone;
 
 service DroneService {
-	rpc StartConfd (Task) returns (Empty);
+	rpc GetInfo(Empty) returns (Info);
+	rpc GetConfdConfig(Empty) returns (ConfdConfig);
+	rpc GetBackendConfig(Empty) returns (BackendConfig);
+	rpc StartConfd (StartConfdRequest) returns (Empty);
+	rpc StopConfd (StopConfdRequest) returns (Empty);
```
Comment (chilianyi) [inline api/drone/drone.proto]: Need request when stop?

File: api/drone/drone.proto [hunk 7]
```diff
@@ -7,24 +7,66 @@ syntax = "proto3";
...
+message StopConfdRequest {
+	ConfdConfig confd_config = 1;
+	BackendConfig backend_config = 2;
+}
+
+message ConfdStatus {
+	string status = 1;
+}
+
+message RegisterCmd {
```
Comment (chilianyi) [inline api/drone/drone.proto]: Can this be changed to `GetRegisterCmdStatusRequest`?
Only cmd_id is needed, cmd_id can be subtask_id. 
```
message GetRegisterCmdStatusRequest {
  string cmd_id =1;
}
```

File: api/frontgate/frontgate.proto [hunk 8]
```diff
@@ -6,39 +6,88 @@ syntax = "proto3";
 
 package openpitrix.frontgate;
 
-service FrontgateService {
+service FrontgateServiceForPilot {
 	rpc GetInfo (Empty) returns (Info);
 	rpc CloseChannel (Empty) returns (Empty);
 
-	rpc StartConfd (Task) returns (Empty);
+	rpc GetConfdInfo(Empty) returns (ConfdInfo);
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: The ConfdInfo is one drone status, Then need a request to sign the drone ip.

File: api/frontgate/frontgate.proto [hunk 9]
```diff
@@ -6,39 +6,88 @@ syntax = "proto3";
...
+	string drone_ip = 1;
+	repeated string frontgate_ips = 2;
+	int32 timeout_second = 3;
 }
 
-message TaskDirective {
+message StopConfdRequest {
 	string drone_ip = 1;
-	string frontgate_id = 2;
+	int32 timeout_second = 3;
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: 2?

File: api/frontgate/frontgate.proto [hunk 10]
```diff
@@ -6,39 +6,88 @@ syntax = "proto3";
...
 
-message Task {
-	string id = 1;
-	string action = 2;
-	string target = 3;
 
-	TaskDirective directive = 4;
+message StartConfdRequest {
+	string drone_ip = 1;
+	repeated string frontgate_ips = 2;
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: frontgate_ip?

File: api/frontgate/frontgate.proto [hunk 11]
```diff
@@ -6,39 +6,88 @@ syntax = "proto3";
...
-	string frontgate_id = 2;
+	int32 timeout_second = 3;
+}
 
-	string command = 3;
-	string command_retry_times = 4;
-	string command_timeout_seconds = 5;
+message RegisterMetadataRequest {
+	string drone_ip = 1;
+	string fg_id = 2;
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: fg unified into frontgate.

File: api/frontgate/frontgate.proto [hunk 12]
```diff
@@ -6,39 +6,88 @@ syntax = "proto3";
...
+	string cmd = 3;
+	int32 timeout_second = 4;
+	int32 retry = 5;
+}
+
+message DeRegisterMetadataRequest {
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: DeregisterMetadataRequest?

File: api/frontgate/frontgate.proto [hunk 13]
```diff
@@ -6,39 +6,88 @@ syntax = "proto3";
...
+	string drone_ip = 1;
+	string fg_id = 2;
+	string fg_ip = 3;
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: Is frontgate_ip needed?

File: api/pilot.proto [hunk 14]
```diff
@@ -51,7 +51,7 @@ message GetSubtaskStatusResponse {
 	repeated SubtaskStatus subtask_status_set = 2;
 }
 
-service PilotManager {
+service PilotService {
 	rpc HandleSubtask (HandleSubtaskRequest) returns (HandleSubtaskResponse) {
 		option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
```
Comment (zheng1) [inline api/pilot.proto]: `HandleSubtask` is internal RPC, no need to define gateway params

File: api/drone/drone.proto [hunk 15]
```diff
@@ -7,24 +7,66 @@ syntax = "proto3";
...
+	BackendConfig backend_config = 2;
+}
+
+message StopConfdRequest {
+	ConfdConfig confd_config = 1;
+	BackendConfig backend_config = 2;
+}
+
+message ConfdStatus {
+	string status = 1;
```
Comment (zheng1) [inline api/drone/drone.proto]: `ConfdStatus ` can include these: uptime, process_id

File: api/drone/drone.proto [hunk 16]
```diff
@@ -7,24 +7,66 @@ syntax = "proto3";
...
+message RegisterCmd {
+	string fg_id = 1;
+	string subtask_id = 2;
+	string cmd = 3;
+}
 
-	string command = 3;
-	string command_retry_times = 4;
-	string command_timeout_seconds = 5;
+message ConfdStatus {
```
Comment (zheng1) [inline api/drone/drone.proto]: duplicate definition

File: api/frontgate/frontgate.proto [hunk 17]
```diff
@@ -6,39 +6,90 @@ syntax = "proto3";
...
+}
+
+message GetConfdInfoRequest {
+	string drone_ip = 1;
 }
 
 message Info {
 	string frontgate_id = 1;
 	string frontgate_service_id = 2;
+	repeated string drone_ip = 3;
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: frontgate_ip?

File: api/drone/drone.proto [hunk 18]
```diff
@@ -7,24 +7,62 @@ syntax = "proto3";
...
+message StartConfdRequest {
+	ConfdConfig confd_config = 1;
+	BackendConfig backend_config = 2;
+}
+
+
+message ConfdStatus {
+	string process_id = 1;
+	string status = 2;
+	google.protobuf.Timestamp up_time = 3;
```
Comment (chilianyi) [inline api/drone/drone.proto]: Should we use status_time?

File: api/frontgate/frontgate.proto [hunk 19]
```diff
@@ -6,39 +6,90 @@ syntax = "proto3";
...
+message ConfdInfo {
+	string drone_ip = 1;
+	string status = 2;
+}
 
-	TaskDirective directive = 4;
+
+message StartConfdRequest {
+	string drone_ip = 1;
+	repeated string frontgate_ip = 2;
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: Is this frontgate_ip needed?

File: api/drone/drone.proto [hunk 20]
```diff
@@ -7,24 +7,62 @@ syntax = "proto3";
...
-	TaskDirective directive = 4;
+service DroneServiceForFrontgate {
+	rpc GetRegisterCmdStatus (GetRegisterCmdStatusRequest) returns (GetRegisterCmdStatusResponse);
 }
 
-message TaskDirective {
+message Info {
 	string drone_ip = 1;
-	string frontgate_id = 2;
+	repeated string cmd = 2;
```
Comment (chilianyi) [inline api/drone/drone.proto]: What is the cmd use for?

File: api/frontgate/frontgate.proto [hunk 21]
```diff
@@ -6,39 +6,88 @@ syntax = "proto3";
...
+	rpc DeregisterMetadata (DeRegisterMetadataRequest) returns (Empty);
 
-	rpc RegisterCmd (Task) returns (Empty);
-	rpc DeregisterCmd (Task) returns (Empty);
+	rpc RegisterCmd (RegisterCmdRequest) returns (Empty);
+	rpc DeregisterCmd (DeRegisterCmdRequest) returns (Empty);
+}
+
+service FrontgateServiceForDrone {
+	rpc GetInfo (Empty) returns (Info);
```
Comment (chilianyi) [inline api/frontgate/frontgate.proto]: What is this rpc use for?  
Drone request to get frontgate info? May be no use.

**Pull request closed without merging**